### PR TITLE
Terms of Service UI Update

### LINKED ISSUE #635: Enhance TOS screen in popup
Terms of service should require you scroll to the bottom to accept.

Should include a link to open in a full window.
Should also include a link to our privacy policy.

### PR DESCRIPTION
**Pull request by 2-am-zzz**
- Updated the UI to reflect our current Terms of Service.
- Updated naming conventions to be more consistent across the application (i.e. changing 'Help' to 'Info')
- Added links in appropriate locations to legal documents on landing page.
- Users must now scroll through the actual ToS to proceed through the app.

Fixes #635

### CODE REVIEW

**Comment by Zanibas** [comment]
Rewriting the current UI test we have to account for the fact that we have to scroll to the bottom of the ToS to activate the button.

File: ui/app/first-time/disclaimer.js [hunk 1]
```diff
@@ -40,16 +40,40 @@ DisclaimerScreen.prototype.render = function () {
...
+          color: blue;
+        }
+
       `),
 
       h('div.markdown', {
+        onScroll: (e) => {
+          var object = e.currentTarget
+          var button = document.getElementById('agree')
+          if ((object.offsetHeight + object.scrollTop + 100 >= object.scrollHeight) && button.disabled) {
```
Comment (flyswatter) [inline ui/app/first-time/disclaimer.js]: This condition is so long it's hard to know what it means. I'd love to see some variables defined that makes it easy to understand.

File: ui/app/first-time/disclaimer.js [hunk 2]
```diff
@@ -67,11 +91,11 @@ DisclaimerScreen.prototype.render = function () {
 
       ]),
 
-      h('button', {
+      h('button#agree', {
```
Comment (flyswatter) [inline ui/app/first-time/disclaimer.js]: Don't use ids, they can really ruin your style sheets because they have ultimate precedence.

**Review by danfinlay — CHANGES_REQUESTED** [review]
There are a number of slight improvements that I think would make this code a little better.

Also, I'd really like to see a mechanism by which users are shown the TOS again when it changes.

File: test/integration/tests.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ QUnit.test('agree to terms', function (assert) {
   wait().then(function() {
 
     var title = app.find('h1').text()
-    assert.equal(title, 'MetaMask', 'title screen')
+    assert.equal(title, 'Terms of Use Version 1', 'title screen')
```
Comment (Zanibas) [inline test/integration/tests.js]: Ignore this: I accidentally committed this and will continue to rewrite the unit test.

File: ui/app/first-time/disclaimer.js (hunk 1 above)
Comment (Zanibas) [inline ui/app/first-time/disclaimer.js]: Rewritten to be more informative and React-y...I don't know what I was doing making all this hard ><

File: ui/app/first-time/disclaimer.js (hunk 2 above)
Comment (Zanibas) [inline ui/app/first-time/disclaimer.js]: Removed ID: the change now properly comes from the state during render-time

**Comment by Zanibas** [comment]
This PR is complete, waiting for final approval.

File: USER_AGREEMENT.md [hunk 4]
```diff
@@ -1,55 +1,181 @@
-# Disclaimer
+# Terms of Use Version 1 #
```
Comment (kumavis) [inline USER_AGREEMENT.md]: i dont really like the "Version 1", keep it simple

@flyswatter thoughts?

File: app/scripts/background.js [hunk 5]
```diff
@@ -8,6 +8,10 @@ const messageManager = require('./lib/message-manager')
 const setupMultiplex = require('./lib/stream-utils.js').setupMultiplex
 const MetamaskController = require('./metamask-controller')
 const extension = require('./lib/extension')
+const fs = require('fs')
+const path = require('path')
+const disclaimer = fs.readFileSync(path.join(__dirname, '..', '..', 'USER_AGREEMENT.md')).toString()
+const stringHash = require('string-hash')
```
Comment (kumavis) [inline app/scripts/background.js]: How big is `string-hash`? We already have a gluttony of crypto libs

**Review by kumavis — APPROVED** [review]
Looks good!

**Comment by flyswatter** [comment]
I agree no version on the tos title.

Also I agree with Aaron on the gluttony of crypto libs. There is not really a good reason to make every user hash the tos every time they start up, it's a waste of CPU and bundle size.

We should do the hashing when we bundle, and just inject the hash as an env var.

- Dan

File: app/scripts/metamask-controller.js [hunk 6]
```diff
@@ -304,6 +341,8 @@ module.exports = class MetamaskController {
     }
   }
 
+
+
```
Comment (flyswatter) [inline app/scripts/metamask-controller.js]: rando newlines

**Review by danfinlay — APPROVED** [review]
Oh I had a "pending review" with some stray style notes.  Review those, fix the ones you want, then LGTM.

File: test/integration/tests.js [hunk 7]
```diff
@@ -3,19 +3,26 @@ QUnit.test('agree to terms', function (assert) {
...
-  // Agree to terms
-  app.find('button').click()
+  app.find('.markdown').prop('scrollTop', 100000000)
 
-  // Wait for view to transition:
+
+  // Agree to terms
   wait().then(function() {
+    app.find('button').click()
+    wait().then(function() {
```
Comment (Zanibas) [inline test/integration/tests.js]: I tried doing it the proper way, but for some reason, it keeps failing.

```
  wait().then(function() {
    app.find('button').click()
  }).then(function() {
    wait()
  }).then(function() {
    var title = app.find('h1').text()
    assert.equal(title, 'MetaMask', 'title screen')

    var buttons = app.find('button')
    assert.equal(buttons.length, 2, 'two buttons: create and restore')

    done()
  })
```

and the tests:

```
not ok 1 Chrome 53.0 - agree to terms
    ---
        actual: >
            MetaMaskTerms of Use
        expected: >
            MetaMask
        stack: >
                at http://localhost:7357/3389/test/integration/tests.js:31:12
        message: >
            title screen
        Log: |
    ...
not ok 2 Firefox 49.0 - agree to terms
    ---
        actual: >
            MetaMaskTerms of Use
        expected: >
            MetaMask
        stack: >
            @http://localhost:7357/3777/test/integration/tests.js:31:5
            
        message: >
            title screen
        Log: |
    ...

1..2
# tests 2
# pass  0
# skip  0
# fail  2
```

Comment (flyswatter) [inline test/integration/tests.js]: In the proper way, you have to `return` a promise within the promise.  So
```
- wait()
+ return wait()
```

Comment (flyswatter) [inline test/integration/tests.js]: https://promisesaplus.com/

Comment (Zanibas) [inline test/integration/tests.js]: It passes now, thanks!

**Pull request merged**